airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/AbstractGscBigQueryUploader.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.uploader;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.JobInfo.WriteDisposition;
+import com.google.cloud.bigquery.LoadJobConfiguration;
+import com.google.cloud.bigquery.TableId;
+import io.airbyte.integrations.destination.bigquery.BigQueryUtils;
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter;
+import io.airbyte.integrations.destination.gcs.GcsDestinationConfig;
+import io.airbyte.integrations.destination.gcs.GcsS3Helper;
+import io.airbyte.integrations.destination.gcs.writer.GscWriter;
+import io.airbyte.protocol.models.AirbyteMessage;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractGscBigQueryUploader<T extends GscWriter> extends AbstractBigQueryUploader<GscWriter> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGscBigQueryUploader.class);
+
+  private final boolean isKeepFilesInGcs;
+  protected final GcsDestinationConfig gcsDestinationConfig;
+
+  AbstractGscBigQueryUploader(final TableId table,
+                              final TableId tmpTable,
+                              final T writer,
+                              final WriteDisposition syncMode,
+                              final GcsDestinationConfig gcsDestinationConfig,
+                              final BigQuery bigQuery,
+                              final boolean isKeepFilesInGcs,
+                              final BigQueryRecordFormatter recordFormatter) {
+    super(table, tmpTable, writer, syncMode, bigQuery, recordFormatter);
+    this.isKeepFilesInGcs = isKeepFilesInGcs;
+    this.gcsDestinationConfig = gcsDestinationConfig;
+  }
+
+  @Override
+  public void postProcessAction(boolean hasFailed) throws Exception {
+    if (isKeepFilesInGcs) {
+      deleteGcsFiles();
+    }
```

Comment:
It looks like the `if` check here missed a negation.

```suggestion
    if (!isKeepFilesInGcs) {
      deleteGcsFiles();
    }
```